### ISSUE TITLE
replace lexicon.jl with documenter.jl for generating docs.

### ISSUE THREAD

**Issue by OkonSamuel** [cstjean/ScikitLearn.jl#71]
According to [Lexicon.jl](https://github.com/MichaelHatherly/Lexicon.jl) Documentation generation using Docile.jl and [Lexicon.jl](https://github.com/MichaelHatherly/Lexicon.jl) is being deprecated in Julia 0.4 and above in favour of [Documenter.jl.](https://github.com/JuliaDocs/Documenter.jl)

**Comment by cstjean** [comment]
Yes! The docs are very much in need of a refresh. In fact, the notebooks should be converted into either Weave.jl or Literate.jl

**Comment by OkonSamuel** [comment]
@cstjean. There is one issue in converting the notebooks to weave or Literate.jl.
- weave only supports Plots.jl plots
- Literate doesn't yet support capturing plots

**Comment by OkonSamuel** [comment]
my plan is to use those notebooks as input and weave them to `markdown` in the doc generation process into a folder. Then finally add them to documenter

**Comment by cstjean** [comment]
That sounds good, too. I've never worked with either weave or literate; I just heard that it was superior to notebooks for docs. Which makes some sense. It's certainly better for `git` tracking.

We could redo the plots in `Plots.jl` too. That's obviously significant work, but it might be good to introduce new users to it rather than `matplotlib`.

**Comment by OkonSamuel** [comment]
@cstjean. OK I guess I will work on that later. But for now can we tag a new release

**Comment by cstjean** [comment]
Sure! Feel free to tag whenever you judge it's appropriate.
